### PR TITLE
Email invitations

### PR DESCRIPTION
**Pull request by rosekamallove**

### CODE REVIEW

File: src/server/api/email/email-sender.ts [hunk 1]
```diff
@@ -121,10 +121,10 @@ export const emailSenderRouter = createTRPCRouter({
   sendUpdatePreview: protectedProcedure
```
Comment (rosekamallove) [inline src/server/api/email/email-sender.ts:121]: Is pure router ki hi koi need nhi hai, alag se kyu banaya hai?

Delete kr de ise

File: src/server/api/email/email.ts [hunk 2]
```diff
@@ -22,12 +22,58 @@ export const emailRouter = createTRPCRouter({
...
+      return email;
+    }),
+
   create: protectedProcedure
     .input(
       z.object({
         subject: z.string(),
         body: z.string(),
         courseId: z.string(),
+        isStyled: z.boolean().optional(),
```
Comment (rosekamallove) [inline src/server/api/email/email.ts:76]: optional() kyu hai yeh? agar select nhi kiya hai toh false bhejo.

File: src/server/api/email/email.ts [hunk 3]
```diff
@@ -22,12 +22,58 @@ export const emailRouter = createTRPCRouter({
     return emails;
```
Comment (rosekamallove) [inline src/server/api/email/email.ts:22]: line:14 pe recipients include mat kar, jab kholega email tab alag se get kr lena

File: src/server/api/email/email.ts [hunk 4]
```diff
@@ -22,12 +22,58 @@ export const emailRouter = createTRPCRouter({
...
+      },
+      orderBy: {
+        updatedAt: "desc",
+      },
+    });
+
+    return draftEmails;
+  }),
+
+  sentMail: protectedProcedure.query(async ({ ctx }) => {
```
Comment (rosekamallove) [inline src/server/api/email/email.ts:41]: getDraftInvitations()

File: src/server/api/email/email.ts [hunk 5]
```diff
@@ -22,12 +22,58 @@ export const emailRouter = createTRPCRouter({
     return emails;
   }),
 
+  draft: protectedProcedure.query(async ({ ctx }) => {
```
Comment (rosekamallove) [inline src/server/api/email/email.ts:25]: getDraftInvitations

File: src/server/api/email/email.ts [hunk 6]
```diff
@@ -71,6 +119,7 @@ export const emailRouter = createTRPCRouter({
               id: ctx.session.user.id,
```
Comment (rosekamallove) [inline src/server/api/email/email.ts:119]: yeh sab kya bakchodi hai bhai? chatGPT se likhwaya hai kya?

creator:{...} wala pura block hatao yaha se

File: src/server/helpers/emailHelper.tsx [hunk 7]
```diff
@@ -45,21 +46,31 @@ const SES_CONFIG = {
...
+    body: body,
+    username: username ?? "",
   };
 
-  const html = template(data);
+  let html=""
+  if (isStyled) {
+   html = render(<EmailInvitation {...data} />);
+  } else {
+    html = template(data);
```
Comment (rosekamallove) [inline src/server/helpers/emailHelper.tsx:72]: bsdk template() aur render() ek hi kaam krre hai. Whatsapp dhang se dekho.

shadowConverter se convert hoga markdown mei, and naye wale blockNote ko phle .md mei convert krna pdega fir .md to .html mei

File: src/pages/creator/dashboard/course/[id]/invitations.tsx [hunk 8]
```diff
@@ -1,601 +1,798 @@
...
-//         </AnimatedSection> */}
-//       </div>
-//     </>
-//   );
-// };
-
-// export default Index;
-// Index.getLayout = CourseNestedLayout;
-
 /* eslint-disable @next/next/no-img-element */
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/course/[id]/invitations.tsx:1]: yeh kyu hai yaha?

File: src/pages/creator/dashboard/course/[id]/invitations.tsx [hunk 9]
```diff
@@ -1,601 +1,798 @@
...
+import { useSession } from "next-auth/react";
+
+interface Email {
+  subject: string;
+  body: string;
+  id: string;
+  isStyled: boolean;
+}
+
+interface Recipient {
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/course/[id]/invitations.tsx:38]: alag se banane ki zruat nhi hai use `RouterOutputs`

File: src/pages/creator/dashboard/course/[id]/invitations.tsx [hunk 10]
```diff
@@ -1,601 +1,798 @@
...
-//             <PlusCircleIcon className="ml-1 w-5" />
-//           </button>
-//         )}
-//       </div>
-//     </form>
-//   );
-return(
-  <h1>j j</h1>
-)
+  const {
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/course/[id]/invitations.tsx:61]: sari kyu chiye jab alag alag banayi hai toh?

File: src/pages/creator/dashboard/course/[id]/invitations.tsx [hunk 11]
```diff
@@ -1,601 +1,798 @@
...
+
+  const { data: importedAudienceData, isLoading: isImpAudLoading } =
+    api.creator.audience.getImportedAudience.useQuery();
+
+  const { mutateAsync: createEmail, isLoading: createEmailLoader } =
+    api.email.create.useMutation();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { data: draftEmail } = api.email.draft.useQuery();
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/course/[id]/invitations.tsx:104]: alag alag component banana pdega iske liye aur unke andar fetch honge yeh wrna alag alag krne ka kya fayeda?

File: src/pages/creator/dashboard/course/[id]/invitations.tsx [hunk 12]
```diff
@@ -1,601 +1,798 @@
...
+                    )}
+                  </button>
+                </div>
+              </Dialog.Panel>
+            </Transition.Child>
+          </div>
+        </div>
+      </Dialog>
+    </Transition>
+  );
```
Comment (rosekamallove) [inline src/pages/creator/dashboard/course/[id]/invitations.tsx:795]: Kuch jada hi lambi file ni hori hai?

File: src/server/api/email/email.ts (hunk 3 above)
Comment (skinnyiam) [reply src/server/api/email/email.ts:22]: ye getAll vala router maine nhi bananya hai pehle se bana tha

File: src/server/api/email/email.ts (hunk 6 above)
Comment (skinnyiam) [reply src/server/api/email/email.ts:119]: bro ye sab router maine nhi banaya hai pehle se bana tha i am just using it , hata de rha main

File: src/pages/creator/dashboard/course/[id]/invitations.tsx (hunk 12 above)
Comment (skinnyiam) [reply src/pages/creator/dashboard/course/[id]/invitations.tsx:795]: ho gaya pehle ban jaye ache se refactor kar dunga

**Pull request merged**